firewood/src/db.rs

```diff
@@ -406,7 +406,8 @@ impl DbRev<MutStore> {
 }
 
 impl From<DbRev<MutStore>> for DbRev<SharedStore> {
-    fn from(value: DbRev<MutStore>) -> Self {
+    fn from(mut value: DbRev<MutStore>) -> Self {
+        value.flush_dirty();
```

Comment:
why do w need to flush dirty here?

Reply:
Without this change, it would be possible to forget to flush the MutStore before consuming it and converting it to a SharedStore. This could result in deferring those writes to the underlying unwriteable shared store.

This only happens once the `DbRev` is finalized and ready for sharing, so at this point we're sure there won't be any more writes to the store. Seems like a great time to flush it.